services/api/src/migrations/2-harbor/harborSync.ts

```diff
@@ -5,33 +5,66 @@ import createHarborOperations from '../../resources/project/harborSetup';
 import projectHelpers from '../../resources/project/helpers';
 import { query, prepare } from '../../util/db';
 
+const lagoonHarborRoute = R.compose(
+  R.defaultTo('http://172.17.0.1:8084'),
+  R.find(R.test(/harbor-nginx/)),
+  R.split(','),
+  R.propOr('', 'LAGOON_ROUTES'),
+)(process.env);
+
 (async () => {
   const sqlClient = getSqlClient();
   const harborClient = createHarborOperations(sqlClient);
 
   logger.info('Syncing Harbor projects with Lagoon projects');
+
   // Get a list of all Lagoon projects
   const projects = await projectHelpers(sqlClient).getAllProjects();
+  const timestamp = Math.floor(Date.now() / 1000)
   for(var i=0; i < projects.length; i++) {
     var project = projects[i];
 
-    // Get project's env vars
-    var envVars = await query(sqlClient, `SELECT * FROM env_vars WHERE project='${project.id}';`);
-    var hasVars = false;
+    // Get project's env vars from db
+    var envVars = await query(
+      sqlClient,
+     `SELECT * 
+        FROM env_vars
+        WHERE project='${project.id}'
+        AND scope = 'internal_container_registry';`
+    );
+    var hasURL = false;
+    var hasUser = false;
+    var hasPass = false;
+    var localLagoonURL = "";
 
     // Check for Harbor variables
     for (var j = 0; j < envVars.length; j++) {
-      if (envVars[j].name == "INTERNAL_REGISTRY_PASSWORD") {
-        hasVars = true;
+      if (envVars[j].name == "INTERNAL_REGISTRY_URL") {
+        hasURL = true;
+        localLagoonURL = envVars[j].value
+      } else if (envVars[j].name == "INTERNAL_REGISTRY_USERNAME") {
+        hasUser = true;
+      } else if (envVars[j].name == "INTERNAL_REGISTRY_PASSWORD") {
+        hasPass = true;
       }
     };
 
-    if (! hasVars) {
+    // Filter for projects using this Lagoon's Harbor route
+    if (localLagoonURL == lagoonHarborRoute) {
+      logger.info('Lagoon project: ', project.name, " is using this Lagoon's Harbor; syncing new robot account for it.");
+      await harborClient.addProject(project.name, project.id, `${project.name}-${timestamp}`);
```

Comment:
I don't understand why we can run `addProject` with a robotName injected as attribute and not, as both cases you do the same:
generate the robot based on the timestamp. Just first time you do it directly in `harborSync.ts` (for the case of `localLagoonURL == lagoonHarborRoute`) and the second time for the case of `localLagoonURL == ""` it happens directly within `harborClient.addProject`.

In my understanding we could always generate the robot name in `harborClient.addProject` no?

Reply:
This convention is used here so that we can continue to use the same `harborClient.addProject` function both here in the migration script and in the kubernetesBuildDeploy system. The k8sBuildDeploy system will pass in a different value for the initially created robot name, so this supports both use cases.

Reply:
> The k8sBuildDeploy system will pass in a different value for the initially created robot name

can you show me where this happens?

Reply:
I am entirely incorrect about this. :(  That functionality would be here, if it existed: https://github.com/amazeeio/lagoon/blob/b7f2525a1bf893f15c1bbc3929ecbc7b5c890b11/services/api/src/resources/project/resolvers.js#L359

I'll update this accordingly so that it always generates a new robot name.

Reply:
👍